### PR TITLE
T070 Add New Turtle export and date range filtering

### PR DESCRIPTION
**Pull request by LizzzzHFFF**
## Summary

This PR introduces support for exporting **New Turtle** observations and adds date range filtering to the export workflow.

A turtle is considered a **New Turtle** when:

- The observation status is **Initial Sighting** or **Initial Nesting**
- The turtle has no previous observations recorded in WAMTRAM

The export can now be filtered by observation date range, allowing users to generate New Turtle exports for a specific reporting period.

### New Turtle Export

- Added a dedicated **New Turtle** export option.
- Export returns observations that meet the New Turtle business rules.

### Date Range Filtering

- Added observation date range filtering to the export.
- Ensured export results are limited to the selected start and end dates.
- Improved handling of date boundaries to prevent observations outside the selected period from being included.

### Query Updates

- Added logic to identify turtles with no prior observations.
- Applied date range filtering to the New Turtle export query.
- Preserved existing export formats and column structure.

### New Turtle Validation

Verified exported records:

- Have an observation status of **Initial Sighting** or **Initial Nesting**
- Have no previous observations recorded in the system
- Are correctly classified as New Turtles

### Export Validation

Tested using:

```text
01/12/2009 - 10/12/2009
```

Verified that:

- New Turtle records are returned correctly
- Records outside the selected date range are excluded
- CSV and Excel exports generate successfully
- Export results are consistent across multiple runs

## Result

✅ Added New Turtle export functionality

✅ Added date range filtering support

✅ Export results respect selected observation dates

✅ Existing export functionality remains unaffected

### CODE REVIEW

File: wamtram2/views.py [hunk 1]
```diff
@@ -1990,8 +2015,19 @@ def _get_date_range(self, date_from, date_to):
...
             return None, None
 
-        start_date = timezone.make_aware(datetime.combine(datetime.strptime(date_from, "%Y-%m-%d").date(), time.min))
-        end_date = timezone.make_aware(datetime.combine(datetime.strptime(date_to, "%Y-%m-%d").date(), time.max))
+        #start_date = timezone.make_aware(datetime.combine(datetime.strptime(date_from, "%Y-%m-%d").date(), time.min))
+        #end_date = timezone.make_aware(datetime.combine(datetime.strptime(date_to, "%Y-%m-%d").date(), time.max))
+        start_date = datetime.combine(
+            datetime.strptime(date_from, "%Y-%m-%d").date(),
+            time.min,
+            tzinfo=datetime_timezone.utc,
```
Comment (RickWangPerth) [inline wamtram2/views.py]: **P1: Date range is interpreted as UTC rather than the application's Perth local date.**

`TIME_ZONE` is `Australia/Perth`, but `_get_date_range()` builds UTC boundaries. Selecting a local business date therefore excludes its first eight AWST hours and includes the next day's first eight hours. Please use `timezone.make_aware(...)` with the active Perth timezone, and add boundary tests around midnight / 08:00 AWST.

File: wamtram2/export_helpers.py [hunk 2]
```diff
@@ -0,0 +1,1331 @@
...
+    ):
+        context["identifications"][
+            identification.observation_id
+        ].append(identification)
+
+    return context
+
+def get_processed_export_row(entry, context):
+
+    observation = entry
```
Comment (RickWangPerth) [inline wamtram2/export_helpers.py]: **P1: The fully prefetched observations are never used when building processed rows.**

`build_processed_export_context()` populates `context["observations"]`, but `get_processed_export_row()` assigns `observation = entry`. The related-object access below therefore falls back to per-row lazy queries. Please resolve it from `context["observations"]` with a safe fallback, or add the same relations to the primary queryset.

File: wamtram2/views.py [hunk 3]
```diff
@@ -2343,283 +2437,268 @@ def export_data(self, request):
...
-                    d_list = damages_dict.setdefault(o_id, [])
-                    if len(d_list) < 2:
-                        d_part = d.body_part_id or ""
-                        d_code = d.damage_code_id or ""
-                        d_list.append((str(d_part), str(d_code)))
-
+            summary_dict = {
+                s.observation_id: s
+                for s in TrvObservationSummary.objects.filter(
+                    observation_id__in=obs_ids
```
Comment (RickWangPerth) [inline wamtram2/views.py]: **P1: Large exports still send every observation ID in one `IN (...)` query.**

The `TRV_OBSERVATION_SUMMARY` lookup here is not chunked, so an export over SQL Server's parameter limit fails despite the new chunking helper elsewhere. Please fetch summary rows in safe chunks as well.

File: wamtram2/export_helpers.py [hunk 4]
```diff
@@ -0,0 +1,1331 @@
...
+    ),
+}
+    
+    return [format_export_value(values.get(header)) for header in PROCESSED_EXPORT_HEADERS]
+
+
+def _safe_queryset(queryset):
+    try:
+        return list(queryset)
+    except DatabaseError:
```
Comment (RickWangPerth) [inline wamtram2/export_helpers.py]: **P1: Database failures are silently converted into empty export data.**

`_safe_queryset()` returns `[]` for every `DatabaseError`, so a failed related-data query can produce a successful but incomplete export. Please surface the failure to the user and log it; only handle a known, recoverable condition explicitly.

File: wamtram2/export_helpers.py [hunk 5]
```diff
@@ -0,0 +1,1331 @@
...
+    if isinstance(value, date):
+        return value.strftime("%m/%d/%Y")
+
+    if isinstance(value, bool):
+        return "True" if value else "False"
+
+    if value is None:
+        return ""
+
+    return str(value)
```
Comment (RickWangPerth) [inline wamtram2/export_helpers.py:155]: **P2: Escape formula-leading user text before writing CSV/XLSX cells.**

Free-text fields such as comments are passed through unchanged into both CSV and XLSX output. A value starting with `=`, `+`, `-`, or `@` can be interpreted as a spreadsheet formula when the recipient opens the export. Please sanitize formula-leading text or explicitly write it as text.

**Review by RickWangPerth — COMMENTED** [review]
Please also add focused unit tests for Perth date boundaries, New Turtle first/non-first observations, organisation scoping, CSV/XLSX headers and populated values, large SQL Server-sized exports, and spreadsheet formula-leading text.

Code quality: `git diff --check` currently reports trailing whitespace and a final blank line. Please also remove duplicate/unused imports and helpers, commented-out legacy code, and the duplicated field-row assembly so there is one tested row-building path.

File: wamtram2/views.py (hunk 1 above)
Comment (LizzzzHFFF) [reply wamtram2/views.py]: Thanks Rick, I've worked through the review feedback.

Updated export date filtering to use Perth-local date boundaries rather than UTC boundaries, which aligns filtering behaviour with how users select business dates.
Updated exports to use Perth-local observation date/time values for both Field Entry and Processed exports.

File: wamtram2/export_helpers.py (hunk 2 above)
Comment (LizzzzHFFF) [reply wamtram2/export_helpers.py]: Updated processed exports to use the prefetched observation objects from the export context instead of falling back to the original row objects.

File: wamtram2/views.py (hunk 3 above)
Comment (LizzzzHFFF) [reply wamtram2/views.py]: Updated observation summary loading to use chunked queries, consistent with the other export lookups and avoiding SQL Server parameter limit issues on large exports.

File: wamtram2/export_helpers.py (hunk 4 above)
Comment (LizzzzHFFF) [reply wamtram2/export_helpers.py]: Removed silent database error swallowing so export failures are surfaced rather than producing incomplete export files.

File: wamtram2/export_helpers.py (hunk 5 above)
Comment (LizzzzHFFF) [reply wamtram2/export_helpers.py:155]: Added spreadsheet formula escaping for values beginning with =, +, -, or @.

**Comment by LizzzzHFFF** [comment]
Cleaned up unused variables/helpers and resolved Ruff findings. ruff check now passes cleanly and git diff --check is clean.

I have not added the suggested focused test coverage in this PR. The functional issues and code quality items have been addressed, but dedicated tests for Perth date boundaries, New Turtle first/non-first observations, organisation scoping, export headers/values, and large export scenarios would be a follow-up piece of work.
